test/components/form/submission.spec.js

```diff
@@ -184,4 +187,36 @@ describe('FormSubmission', () => {
           }));
     });
   });
+
+  describe('session timeout', () => {
+    beforeEach(() => {
+      mockLogin();
+    });
+
+    it('does not auto log out', async () => {
+      testData.extendedForms.createPast(1, { xmlFormId: 'a', webformsEnabled: true });
+      const session = testData.sessions.createNew({ expiresAt: '1970-01-01T00:05:00Z' });
+
+      const clock = sinon.useFakeTimers();
+
+      await load('/projects/1/forms/a/submissions/new', mountOptions({
+        container: {
+          requestData: testRequestData([], {
+            session
+          }),
+        }
+      }))
+        .complete()
+        .request(() => {
+          clock.tick('24:00:00');
+        })
+        .respondWithData(() => '(v2024.1.2-sha)')
+        .testRequests([
+          { url: '/version.txt' }
+        ])
+        .afterResponses((app) => {
+          app.vm.$container.requestData.session.dataExists.should.be.true;
```

Comment:
Could you also assert that the route path hasn't changed? In other words, that the user wasn't redirected to /login.